squape/object_tree.py

```diff
@@ -0,0 +1,154 @@
+        selector (dict, optional): The selector is a dictionary of properties,
+        that must match for objects to be included into the result.
+        Defaults to {}, which means all objects pass the verification.
+
+    Returns:
+        tuple: the result of search among the object tree.
+
+    Examples:
+        find(object)
+        find(object, {'type' : ToolBar})
+        find(object, max_depth=5)
+        find(object, {'visible' : True}, max_depth=3)
+    """
+    if max_depth is None:
+        max_depth = math.inf
+    if _depth >= max_depth:
+        return []
+    if selector is None:
+        selector = {}
+    children = list(object.children(obj))
+    children_count = len(children)
+
+    for index, child in enumerate(children):
+        if index == children_count:
+            break
+        children.extend(find(child, max_depth=max_depth, _depth=_depth + 1))
+
+    if _depth == 0:
+        filtered_children = filter(lambda x: _filter_by_selector(x, selector), children)
+        return tuple(filtered_children)
+    else:
+        return children
+
+
+def find_parent(obj, selector: dict = None):
+    """Find the first parent that matches the selector,
+    among all the parent objects up to the root.
+
+    Args:
+        obj (Squish object): A parent node.
+        selector (dict, optional): The selector is a dictionary of properties,
+        that must match for objects to be included into the result.
+        Defaults to {}, which means all objects pass the verification.
+
+    Returns:
+        Squish object / None: The parent object that matches the selector.
+        None if such a parent does not exist.
+
+    Examples:
+        find_parent(obj, {'type' : 'MyContainerType'})
+    """
+    if selector is None:
+        selector = {}
+
+    if object.parent(obj) is None:
+        return None
+    
+    if _filter_by_selector(object.parent(obj), selector):
+        return object.parent(obj)
+        
+    return find_parent(object.parent(obj), selector)
+
+
+def siblings(obj, selector: dict = None):
+    """Find all the siblings (direct children of the parent) object
+    for a given object, that match the selector.
+
+    Args:
+        obj (Squish object): A parent node.
+        selector (dict, optional): The selector is a dictionary of properties,
+        that must match for objects to be included into the result.
+        Defaults to {}, which means all objects pass the verification.
+
+    Returns:
+        tuple: the result of search among the object tree.
+
+    Examples:
+        siblings(object)
+        siblings(object, {'enabled' : True})
+    """
+    if selector is None:
+        selector = {}
+    parent = object.parent(obj)
+
+    if parent is None:
+        return None
+    else:
+        siblings = list(object.children(parent))
+        siblings.remove(obj)
+        return tuple(filter(lambda x: _filter_by_selector(x, selector), siblings))
+
+
+def _filter_by_selector(obj, selector: dict) -> bool:
```

Comment:
1. same comments to naming, parameter type, and docstrings as previously
2. the name of the function suggests that some kind of filtering is done here which is not the case I believe